monai/losses/unified_focal_loss.py

```
@@ -74,167 +76,219 @@ def forward(self, y_pred: torch.Tensor, y_true: torch.Tensor) -> torch.Tensor:
+            if n_pred_ch == 1:
+                warnings.warn("single channel prediction, softmax=True ignored.")
+            else:
+                y_pred_act = torch.softmax(y_pred, dim=1)
 
-        n_pred_ch = y_pred.shape[1]
         if self.to_onehot_y:
-            if n_pred_ch == 1:
+            if n_pred_ch == 1 and not self.sigmoid:
                 warnings.warn("single channel prediction, `to_onehot_y=True` ignored.")
-            else:
+            elif n_pred_ch > 1 or self.sigmoid:
```

Ruff (0.14.2)

No explicit `stacklevel` keyword argument found

Set `stacklevel=2`

(B028)